app/src/main/java/org/listenbrainz/android/viewmodel/DashBoardViewModel.kt

```diff
@@ -94,24 +98,147 @@ class DashBoardViewModel @Inject constructor(
 
     fun markPermissionAsRequested(permission: PermissionEnum) {
         viewModelScope.launch(ioDispatcher) {
-            val permissions = appPreferences.requestedPermissionsList.getFlow().firstOrNull()?.toMutableList()
+            val permissions =
+                appPreferences.requestedPermissionsList.getFlow().firstOrNull()?.toMutableList()
             if (permissions != null && !permissions.contains(permission.permission)) {
                 permissions.add(permission.permission)
                 appPreferences.requestedPermissionsList.set(permissions)
             }
         }
     }
 
-    fun markOnboardingComplete(){
+    fun getListeningApps(context: Activity) {
```

Comment:
A few things here:

1. This function can be separated into a util function and function which updates state inside this view-model. The util  function can be placed in the utils file.

2. This function is named `getListeningApps` but it does not return anything. Nomenclature is important in a codebase. Please name it `updateListeningApps` to avoid confusion.

3. The new util function can be an extension function with a return type as follows:
`Context.getListenableApps(): <List<ApplicationInfo>>`